### ISSUE TITLE
It was not possible to do the requested undo operation (from server log)

### ISSUE THREAD

**Issue by drmDev** [magefree/mage#3708]
```
GameImpl.playPriority
java.lang.UnsupportedOperationException: It was not possible to do the requested undo operation (bookmark 3 does not exist) 
context: {2}{R}{G}: Until end of turn, {this} becomes a 3/3 red and green Elemental creature with "Whenever this creature attacks, put a +1/+1 counter on it." It's still a land.
	at mage.game.GameImpl.restoreState(GameImpl.java:637)
	at mage.players.PlayerImpl.restoreState(PlayerImpl.java:1144)
	at mage.players.PlayerImpl.playAbility(PlayerImpl.java:1108)
	at mage.players.PlayerImpl.activateAbility(PlayerImpl.java:1180)
	at mage.player.human.HumanPlayer.activateAbility(HumanPlayer.java:1350)
	at mage.player.human.HumanPlayer.activateAbility(HumanPlayer.java:1364)
	at mage.player.human.HumanPlayer.priority(HumanPlayer.java:769)
	at mage.game.GameImpl.playPriority(GameImpl.java:1274)
	at mage.game.turn.Step.priority(Step.java:87)
	at mage.game.turn.Phase.playStep(Phase.java:210)
	at mage.game.turn.Phase.play(Phase.java:115)
	at mage.game.turn.Turn.play(Turn.java:153)
	at mage.game.GameImpl.playTurn(GameImpl.java:822)
	at mage.game.GameImpl.play(GameImpl.java:731)
	at mage.game.GameImpl.start(GameImpl.java:695)
	at mage.server.game.GameWorker.call(GameWorker.java:60)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by drmDev** [comment]
Possibly a more general issue with undo or rollbacks?

**Comment by LevelX2** [comment]
Sure, this is no new problem. It can also happen if you cancel casting/activating spells/abilities, because this is also handled with the state reloading. 
Nice would be to have a reproducable scenario, so I can debug the reason.

**Comment by JayDi85** [comment]
Same as #925. Most popular situations with cast cancelling (from de-server's logs) must be fixed by #6937. But more situations are possible in theory (see #6941 for details). That issue can be closed now.
